src/Parsers/GetObject/MultipleStreamed.php

```diff
@@ -1,30 +1,39 @@
-<?php namespace PHRETS\Parsers\GetObject\Multiple;
+<?php namespace PHRETS\Parsers\GetObject;
 
-use GuzzleHttp\Psr7\Response;
 use PHRETS\Http\Response as PHRETSResponse;
-use Illuminate\Support\Collection;
-use PHRETS\Parsers\GetObject\Single;
 
-class Streamed
+class MultipleStreamed
 {
-    // in bytes
-    // TODO: parameterize this
-    private const CHUNK_SIZE = 20 * 1000 * 1000;
+    // 20MB
+    const DEFAULT_CHUNK_SIZE = 20000000;
 
-    public function parse(PHRETSResponse $response) : \Generator
+    public function parse(PHRETSResponse $response, $chunkSize = null)
     {
         if (!$response->getBody()) {
             return;
         }
 
-        $boundary = $this->determineBoundary($response->getHeader('Content-Type'));
+        $chunkSize = $chunkSize ?: self::DEFAULT_CHUNK_SIZE;
+
         $singleObjectParser = new Single();
 
-        $firstChunk = true;
+        $isFirstChunk = true;
         $inProgressSegment = '';
 
+        $boundary = $this->determineBoundary($response->getHeader('Content-Type'));
+        $boundaryRegex = "/(\r\n|^)--{$boundary}-?-?(\r\n|$)/";
+        // dummy preg_match() to test if the regex is valid
```

Comment:
:+1:

Reply:
If it's not clear, this is to avoid waiting until we've loaded in the first chunk to test the validity of the regex